test/core/tsi/s2a/s2a_test_util.cc

```diff
@@ -323,3 +219,102 @@ bool check_record_empty_plaintext(uint16_t ciphersuite,
   }
   return true;
 }
+
+void random_array(uint8_t* bytes, size_t length) {
+  if (length == 0) return;
+  GPR_ASSERT(bytes != nullptr);
+  for (size_t i = 0; i < length; i++) {
+    bytes[i] = static_cast<uint8_t>(rand() % 256);
+  }
+}
+
+void send_random_message(size_t message_size, s2a_crypter* out_crypter,
+                         s2a_crypter* in_crypter) {
+  GPR_ASSERT(out_crypter != nullptr && in_crypter != nullptr);
+  GPR_ASSERT(out_crypter != in_crypter);
+  size_t max_record_overhead;
+  char* error_details = nullptr;
+  grpc_status_code max_overhead_status = s2a_max_record_overhead(
+      out_crypter, &max_record_overhead, &error_details);
+  if (max_overhead_status != GRPC_STATUS_OK) {
+    gpr_log(GPR_ERROR, "%s", error_details);
+    gpr_free(error_details);
+  }
+  GPR_ASSERT(max_overhead_status == GRPC_STATUS_OK);
+  GPR_ASSERT(error_details == nullptr);
+  GPR_ASSERT(message_size <= SSL3_RT_MAX_PLAIN_LENGTH + max_record_overhead);
+  std::vector<uint8_t> message(message_size, 0);
+  random_array(message.data(), message_size);
+
+  size_t record_allocated_size = message_size + max_record_overhead;
+  std::vector<uint8_t> record(record_allocated_size, 0);
+  size_t record_size;
+
+  grpc_status_code encrypt_status =
+      s2a_encrypt(out_crypter, message.data(), message_size, record.data(),
+                  record.size(), &record_size, &error_details);
+  GPR_ASSERT(encrypt_status == GRPC_STATUS_OK);
+  GPR_ASSERT(error_details == nullptr);
+  GPR_ASSERT(record_size == expected_message_size(message_size));
+
+  size_t plaintext_allocated_size;
+  grpc_status_code plaintext_status = s2a_max_plaintext_size(
+      in_crypter, record_size, &plaintext_allocated_size, &error_details);
+  if (plaintext_status != GRPC_STATUS_OK) {
+    gpr_log(GPR_ERROR, "%s", error_details);
+    gpr_free(error_details);
+  }
+  GPR_ASSERT(plaintext_status == GRPC_STATUS_OK);
+  GPR_ASSERT(error_details == nullptr);
+  std::vector<uint8_t> plaintext(plaintext_allocated_size, 0);
+  size_t plaintext_size;
+  s2a_decrypt_status decrypt_status =
+      s2a_decrypt(in_crypter, record.data(), record_size, plaintext.data(),
+                  plaintext.size(), &plaintext_size, &error_details);
+  GPR_ASSERT(decrypt_status == OK);
+  GPR_ASSERT(error_details == nullptr);
+  GPR_ASSERT(plaintext_size == message_size);
+  plaintext.resize(plaintext_size);
+  GPR_ASSERT(plaintext == message);
+}
+
+grpc_status_code create_random_crypter_pair(uint16_t ciphersuite,
```

Comment:
Same comment here. We can define const secrets.

Reply:
Same as above.

Reply:
I can't seem to find where "above" refers to. Can you please point me to that? Thanks!

Reply:
Sorry - I was unclear. I removed this test entirely, and wanted to check if you're OK with the new testing framework (i.e. have the `s2a_test_data.h` file with hardcoded messages). If you're OK with the new framework, I'll add in the hardcoded version of this test (maybe as a separate PR, to avoid clutter).

Reply:
SG, the tests LGTM and a separate PR is fine.